### PR TITLE
Feat/challenge 02

### PR DESCRIPTION
**Pull request by Daniel94y1**

### CODE REVIEW

**Comment by Daniel94y1** [comment]
Merge branch feat/challenge_02 to main

File: js/exercise-00.js [hunk 1]
```diff
@@ -0,0 +1,38 @@
...
+   - Boolean
+   - Null
+   - Undefined
+
+   Log out each variable to the terminal at the end.
+*/
+
+//Declare variables
+
+        const stringVariable = "Hello cats!";
```
Comment (tictools) [inline js/exercise-00.js]: Unnecessary spaces.

```suggestion
const stringVariable = "Hello cats!";
```

FIX: remove initial spaces

File: js/exercise-02.js [hunk 2]
```diff
@@ -0,0 +1,36 @@
...
+//console.log("Sound:", sound);
+
+// Explanation:
+// It gives an error since you have to declare and initialize a const variable at the same time.
+// it cannot be done separately, since the const variable's value is not updated.
+
+//Solution:
+
+const soundConst="Moo!";
+console.log("Sound:", soundConst);
```
Comment (tictools) [inline js/exercise-02.js]: FIX: add empty line at the end fo the file

File: js/exercise-03.js [hunk 3]
```diff
@@ -0,0 +1,17 @@
...
+function getTypeOf(value) {
+    return typeof value;
+}
+
+//Output function
+console.log(getTypeOf("Hello, World!")); // Devuelve: string
+console.log(getTypeOf(42)); // Devuelve: number
+console.log(getTypeOf(true)); // Devuelve: boolean
+console.log(getTypeOf(null)); // Devuelve: object (¡esto es un error en JavaScript!)
+console.log(getTypeOf(undefined)); // Devuelve: undefined
```
Comment (tictools) [inline js/exercise-03.js]: FIX: add empty line at the end fo the file

File: js/exercise-03.js [hunk 4]
```diff
@@ -0,0 +1,17 @@
...
+    2. Log the result of any function call.
+*/
+
+//Declare function
+function getTypeOf(value) {
+    return typeof value;
+}
+
+//Output function
+console.log(getTypeOf("Hello, World!")); // Devuelve: string
```
Comment (tictools) [inline js/exercise-03.js]: FIX: code comments and documentation should be written in 🇬🇧

File: js/exercise-04.js
Comment (tictools) [inline js/exercise-04.js:1]: Exercise not done

File: js/exercise-04.js [hunk 5]
```diff
@@ -0,0 +1,34 @@
...
+    // In this case, false is falsy, so it short-circuits the evaluation and doesn't proceed to the right operand.
+
+// true || console.log("hello friend");
+    // This line doesn't log anything. It works this way because the || (OR) operator checks if the left operand is truthy.
+    // In this case, true is truthy, so it short-circuits the evaluation and doesn't proceed to the right operand.
+
+// false || console.log("bye friend");
+    // This line logs "bye friend" to the console. It works this way because the || operator checks if the left operand is falsy.
+    // In this case, false is falsy, so it proceeds to the right operand, which is console.log("bye friend"), and it logs "bye friend" to the console. 
+    // The result of the console.log statement doesn't affect the overall result of the || operation.
```
Comment (tictools) [inline js/exercise-04.js]: FIX: add empty line at the end fo the file

File: js/exercise-03.js [hunk 6]
```diff
@@ -0,0 +1,23 @@
...
+console.log(getTypeOf(70)); 
+console.log(getTypeOf(true)); 
+console.log(getTypeOf(null)); 
+console.log(getTypeOf(undefined)); 
+
+//Explanation
+// The function takes the value as a parameter and returns the type of the collected value.
+//The problem is that when it collects the null value, it returns the value type object.
+// This happens because null is internally represented with a pointer that points to the special memory address that represents "absence of value."
+//This address was interpreted as the type of object, which is why this function returns "object"
```
Comment (tictools) [inline js/exercise-03.js]: FIX: add empty line at the end fo the file

File: js/exercise-03.js [hunk 7]
```diff
@@ -0,0 +1,23 @@
...
+    2. Log the result of any function call.
+*/
+
+//Declare function
+function getTypeOf(value) {
+    return typeof value;
+}
+
+//Output function
+console.log(getTypeOf("Hello cats!"));
```
Comment (tictools) [inline js/exercise-03.js]: Which is the output of these type evaluations?

```suggestion
console.log(getTypeOf("Hello cats!"));  // string
```

FIX: add logs output

File: js/exercise-03.js [hunk 8]
```diff
@@ -0,0 +1,23 @@
...
+}
+
+//Output function
+console.log(getTypeOf("Hello cats!")); 
+console.log(getTypeOf(70)); 
+console.log(getTypeOf(true)); 
+console.log(getTypeOf(null)); 
+console.log(getTypeOf(undefined)); 
+
+//Explanation
```
Comment (tictools) [inline js/exercise-03.js]: This explanation is not necessary.

FIX: remove comment

**Comment by Daniel94y1** [comment]
Complete the revision

**Review by tictools — APPROVED** [review]

**Pull request merged**